table/iterator.go

```diff
@@ -208,6 +213,21 @@ func (itr *Iterator) seekHelper(blockIdx int, key []byte) {
 	itr.err = itr.bi.Error()
 }
 
+func (itr *Iterator) seekFromOffset(blockIdx int, offset int, key []byte) {
+	itr.bpos = blockIdx
+	block, err := itr.t.block(blockIdx)
+	if err != nil {
+		itr.err = err
+		return
+	}
+	itr.bi.setBlock(block)
+	itr.bi.setIdx(offset)
+	if y.CompareKeys(itr.bi.key, key) >= 0 {
+		return
+	}
+	itr.bi.seek(key)
```

Comment:
The version we are looking for may be in the next block, we may not be able to find it in the current block.
So we need to fall back to `itr.seekFrom`.

Reply:
If different version of a key fall into two block, there is a hash index collision.

Reply:
same hash point to different block.

Reply:
Got it.